tests/integration/test_split.py

```diff
@@ -0,0 +1,343 @@
+            # Numpy returns  [array([], dtype=int64)]
+
+    def test_dimensions_hsplit_1(self):
+        ary = num.arange(10)
+        expected_exc = ValueError
+        with pytest.raises(expected_exc):
+            num.hsplit(ary, 1)
+        with pytest.raises(expected_exc):
+            np.hsplit(ary, 1)
+
+    @pytest.mark.parametrize("size", ((0,), (10,), (5, 5)))
+    def test_dimensions_dsplit(self, size):
+        ary = np.random.randint(low=0, high=100, size=size)
+        expected_exc = ValueError
+        with pytest.raises(expected_exc):
+            num.dsplit(ary, 1)
+        with pytest.raises(expected_exc):
+            np.dsplit(ary, 1)
+
+
+def compare_array(a, b):
+    """
+    Compare two array using zip method.
+    """
+    if len(a) != len(b):
+        return False
+    else:
+        for each in zip(a, b):
+            if not np.array_equal(*each):
+                return False
+    return True
+
+
+@pytest.mark.parametrize("size", SIZES, ids=str)
+def test_split(size):
+    a = np.random.randint(low=0, high=100, size=size)
+    axis_list = list(range(a.ndim))
+    axis_list.append(-1)
+    for axis in axis_list:
+        input_arr = []
+        even_div = None
+        uneven_div = None
+
+        if a.shape[axis] > 1:
+            for div in range(1, (int)(math.sqrt(a.shape[axis]) + 1)):
+                if a.shape[axis] % div == 0:
+                    even_div = div
+                else:
+                    uneven_div = div
+                if even_div is not None:
+                    break
+
+        # divisible integer
+        if even_div:
+            input_arr.append(even_div)
+        else:
+            input_arr.append(1)
+
+        if uneven_div:
+            input_arr.append(uneven_div)
+        # indices array which has points
+        # within the target dimension of the src array
+        if a.shape[axis] > 1 and even_div is not None:
+            input_arr.append(list(range(1, a.shape[axis], even_div)))
+        # indices array which has points
+        # out of the target dimension of the src array
+        if even_div is not None:
+            input_arr.append(
+                list(
+                    range(
+                        0,
+                        a.shape[axis] + even_div * np.random.randint(1, 10),
+                        even_div,
+                    )
+                )
+            )
+            input_arr.append(
+                list(
+                    range(
+                        a.shape[axis] + even_div * np.random.randint(1, 10),
+                        0,
+                        -even_div,
+                    )
+                )
+            )
+
+        for input_opt in input_arr:
+            func_num = getattr(num, "split")
+            func_np = getattr(np, "split")
+
+            res_num = func_num(a, input_opt, axis=axis)
+            res_np = func_np(a, input_opt, axis=axis)
+
+            assert compare_array(res_num, res_np)
```

Comment:
This is a very complicated test, it seems to be trying to accommodate a few different scenarios all at once. I'd suggest simplifying (and the test below). I think the best thing would be to split `SIZES` up according to even/uneven and have separate (simpler) test functions for each. Some helper functions might also help here.